src/submission/schema/schema-manager.ts

```diff
@@ -268,20 +282,77 @@ namespace MigrationManager {
       newSchemaVersion,
     );
 
-    await checkDonorDocuments(migration, newTargetSchema);
+    const migration1 = await checkDonorDocuments(migration, newTargetSchema);
+
+    await revalidateOpenSubmissionsWithNewSchema(migration1, newTargetSchema, migration.dryRun);
 
     // close migration
     const updatedMigration = await migrationRepo.getById(migrationId);
     const migrationToClose = _.cloneDeep(updatedMigration) as DictionaryMigration;
+
     if (!migrationToClose) {
       throw new Error('where did the migration go? expected migration not found');
     }
+
     migrationToClose.state = 'CLOSED';
     migrationToClose.stage = 'COMPLETED';
     const closedMigration = await migrationRepo.update(migrationToClose);
     return closedMigration;
   };
 
+  const revalidateOpenSubmissionsWithNewSchema = async (
+    migration: DictionaryMigration,
+    newSchema: SchemasDictionary,
+    dryRun: boolean,
+  ) => {
+    const submissions = await submissionService.operations.findActiveClinicalSubmissions();
+
+    for (const sub of submissions) {
+      if (
+        sub.state == SUBMISSION_STATE.INVALID ||
+        sub.state == SUBMISSION_STATE.INVALID_BY_MIGRATION
+      ) {
+        continue;
+      }
+
+      const checkedSubmission = migration.checkedSubmissions.find(x => {
```

Comment:
Unless I'm misreading, the method provided to find here never returns a value, no? just assigns the `sub._id` to `x._id` for every element in checkedSubmissions.

Reply:
rookie mistake, should been ===

Reply:
Compact way to write this in the future:
`find(x => x._id === sub._id)`